CMakeLists.txt

```diff
@@ -10,6 +10,8 @@ include(GNUInstallDirs)
 option(BUILD_DOCS "Build documentation" FALSE)
 option(ENABLE_TESTS "Enable tests" FALSE)
 
+find_package(Boost REQUIRED COMPONENTS container random filesystem log log_setup program_options thread system unit_test_framework timer)
```

Comment:
why we need move it here?

Reply:
Without it, the configure phase is failing:

```
zkevm-framework> CMake Generate step failed.  Build files cannot be regenerated correctly.
error: builder for '/nix/store/vvvam9v0w98sb8aabvf2vhv616h4h2kz-zkevm-framework.drv' failed with exit code 1;
       last 25 log lines:
       > -- Configuring done (1.1s)
       > CMake Error at tests/libs/assigner_runner/CMakeLists.txt:3 (target_link_libraries):
       >   Target "assigner_runner_test" links to:
       >
       >     Boost::random
       >
       >   but the target was not found.  Possible reasons include:
       >
       >     * There is a typo in the target name.
       >     * A find_package call is missing for an IMPORTED target.
       >     * An ALIAS target is missing.
```

Reply:
we have to fix it carefully.
Different libraries/binaries has different boost dependencies, so need allow way for partial build (only one target) without using useless packages

Reply:
We are propagating the boost dependency now, so it's avoided